File: src/plugin/rsp/rsp.ts
import { Account } from "../../account";
import { Receipt } from "../../receipt";
import { CaptureReceiptEvent } from '../plugin-event'
import { Capture }

export interface Rsp {
  requestId: string,
  event: CaptureReceiptEvent
  payload: CaptureReceiptError | Account | Receipt
}

File: android/src/main/kotlin/com/mytiki/sdk/capture/receipt/capacitor/plugin/rsp/Rsp.kt
/*
 * Copyright (c) TIKI Inc.
 * MIT license. See LICENSE file in root directory.
 */

package com.mytiki.sdk.capture.receipt.capacitor.plugin.rsp

import com.getcapacitor.JSObject
import com.mytiki.sdk.capture.receipt.capacitor.plugin.PluginEvent

open class Rsp(val requestId: String, val event: PluginEvent) {
    open fun toJS(): JSObject = JSObject()
            .put("requestId", this.requestId)
            .put("event", this.event)
}
